OOPHomework14January/src/Store.java

```diff
@@ -0,0 +1,292 @@
+                found = true;
+            }
+        }
+        return found;
+    }
+
+    /**
+     * method to avoid registering a duplicate id
+     *
+     * @param id - the id we want to check
+     * @return true - if the id is already given to another product; and false - otherwise
+     */
+    private boolean duplicateAnimalID(int id) {
+        for (AnimalProduct animalProduct : arrayAnimalProduct) {
+            if (animalProduct.getId() == id) {
+                return false;
+            }
+        }
+        return true;
+    }
+    /**
+     * method to avoid registering a duplicate id
+     *
+     * @param id - the id we want to check
+     * @return true - if the id is already given to another product; and false - otherwise
+     */
+    private boolean duplicateVegetlID(int id) {
+        for (VegetalProduct vegetalProduct : arrayVegetalProduct) {
+            if (vegetalProduct.getId() == id) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * method to avoid registering a duplicate id
+     *
+     * @param id - the id we want to check
+     * @return true - if the id is already given to another product; and false - otherwise
+     */
+    private boolean duplicateID(int id) {
+        return duplicateAnimalID(id)&& duplicateVegetlID(id);
+    }
+
+    /**
+     * method to display all the animal products from the store
+     */
+    public void displayAnimalProducts() {
+        System.out.println("ID\tName\t\t\tQuantity\n------------------------------------");
+        for (AnimalProduct animalProduct : arrayAnimalProduct) {
+            System.out.println(animalProduct.getId() + "\t" +
+                    animalProduct.getProductName() + "\t\t" +
+                    animalProduct.getStock());
+        }
+    }
+
+    /**
+     * method to display all the vegetal products from the store
+     */
+    public void displayVegetalProducts() {
+        for (VegetalProduct vegetalProduct : arrayVegetalProduct) {
+            System.out.println(vegetalProduct.getId() + "\t" +
+                    vegetalProduct.getProductName() + "\t\t" +
+                    vegetalProduct.getStock());
+        }
+    }
+
+    /**
+     * method to display the orders from a specific date
+     */
+    public void displayOrders(String date) {
+        System.out.println("Order placed on " + date + ":\n");
+        System.out.println("ID\tQuantity\n----------------------");
+        for (Order order : arrayOrders) {
+            if (order.getDate().toString().equals(date)) {
+                System.out.println(
+                        order.getId() + "\t" +
+                                order.getQuantity());
+            }
+        }
+    }
+
+
+    /**
+     * method to get the local date
+     */
+    private LocalDate localDate() {
+        Date input = new Date();
+        return (input.toInstant().atZone(ZoneId.systemDefault()).toLocalDate());
```

Comment:
new Date().toInstant()
is the same thing as:
Instant.now()

Also, if you don't have any specific timezone you need and you just want the systemDefault one, then this entire method could be replaced by:
LocalDate.now();